docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java

```diff
@@ -227,10 +232,9 @@ public DockerSlave provision(StreamTaskListener listener) throws IOException, De
 
         logger.println("Launching " + image );
 
-        int numExecutors = 1;
         Node.Mode mode = Node.Mode.NORMAL;
 
-        RetentionStrategy retentionStrategy = new OnceRetentionStrategy(idleTerminationMinutes());
+        RetentionStrategy retentionStrategy = new CloudRetentionStrategy(idleTerminationMinutes());
```

Comment:
Please make the use of CloudRetentionStrategy optional, eg. only use it when multiple executors are configured.